src/aiida/cmdline/utils/decorators.py

```diff
@@ -259,7 +259,10 @@ def wrapper(wrapped, _, args, kwargs):
 
         template = templates.env.get_template('deprecated.tpl')
         width = 80
-        echo.echo(template.render(msg=wrap(message, width - 4), width=width))
+        message_ = message
+        if version is not None:
+            message_ = f'{message_} (this will be removed in v{version})'
```

Comment:
This will fail. As soon as we want to assign `message` it is assumed to be `local`, so the `message` variable that comes from outside is now not accessible, but in the assignment we need it. So one way is to make a new local variable or to mark `message` as `global`, but then we change it outside of the function which might be confusing if the code ever gets extended.

A simple example of this is behavior.
```python
x = 1
def foo():
    x = x
    return x
```
This will cause a `UnboundLocalError: cannot access local variable 'x' where it is not associated with a value. While this works.
````python
x = 1
def foo():
    global x
    x = x
    return x
```

But I agree the code seems weird when one is not aware of this, and it is unintuitive pythonic behavior. I changed the logic a bit to avoid thinking about this.

Reply:
> This will fail.

Are you sure?
The following example seems to suggest otherwise:
```python
In [1]: def deprecated_command(message, version=None):
   ...:     if version is not None:
   ...:         message = f'{message} (this will be removed in v{version})'
   ...:     print(message)
   ...: 

In [2]: deprecated_command('This is a message')
This is a message

In [3]: deprecated_command('This is a message', version=3)
This is a message (this will be removed in v3)
```

Am I missing anything?

Reply:
Ah yeah, I see now. Then it makes sense. Maybe the following change makes it a bit cleaner still:
```python
        message = wrap(message if version is None else message + f' (this will be removed in v{version})')
        echo.echo(template.render(msg=message, width - 4), width=width)
```

Reply:
I could implement this version logic to in VerdiCommand that requires a new initialization parameter `deprecated_version`. I can also put this as an issue so we can get this faster merged and implement the version later as marking a version for `3.0.0` as deprecated is not so urgent.

Reply:
I just realized what you meant that this is in the `deprecated_command` decorator which is more or less "deprecated" itself, as it is just automatically used by the `VerdiCommand`. I agree with you that this doesn't really belong in this PR. Let's just take it out for now and then we can follow up with this later. Probably we need to add an argument to the `command` method to declare this version if we think it is useful. And it can then pass this on to `echo_deprecated`.